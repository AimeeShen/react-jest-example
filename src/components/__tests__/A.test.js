import React, { Component } from 'react';
import { mount } from 'enzyme';

describe('<A />', () => {
  let makeSubject;
  let params;
  let A;

  beforeEach(() => {
    params = {};
    A = require('../A').default;

    makeSubject = () => mount(<A />);
  });

  it('should match latest snapshot', () => {
    const subject = makeSubject();

    expect(subject.html()).toMatchSnapshot();
  });

  it('should call foo when .btn-foo clicked', () => {
    const subject = makeSubject();
    
    subject.instance().foo = jest.fn();
    subject.update();
    subject.find('.btn-foo').simulate('click');
    expect(subject.instance().foo).toHaveBeenCalled();
  });
});
