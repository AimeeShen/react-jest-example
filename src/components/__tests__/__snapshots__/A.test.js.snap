// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<A /> should match latest snapshot 1`] = `"<div><button class=\\"btn-foo\\">click me</button></div>"`;
